modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/io/FileSystem.scala

```diff
@@ -51,8 +51,9 @@ object FileSystem {
       .subflatMap { text =>
         val either = for {
           tsConfig <- Either.catchNonFatal(ConfigFactory.parseString(text)).leftMap(_.getMessage)
-          tsConfig <- Either.catchNonFatal(fallbacks(tsConfig)).leftMap(_.getMessage)
-          parsed <- tsConfig.as[A].leftMap(_.show)
+          resolved <- Either.catchNonFatal(tsConfig.resolve()).leftMap(e => s"Can't resolve config: ${e.getMessage}")
+          withFallbacks <- Either.catchNonFatal(fallbacks(resolved)).leftMap(_.getMessage)
+          parsed <- withFallbacks.as[A].leftMap(_.show)
```

Comment:
This might be controversial, but if it was me I would write it like this, shadowing the `tsConfig` parameter.

```
    tsConfig <- Either.catchNonFatal(ConfigFactory.parseString(text)).leftMap(_.getMessage)
    tsConfig <- Either.catchNonFatal(tsConfig.resolve()).leftMap(e => s"Can't resolve config: ${e.getMessage}")
    tsConfig <- Either.catchNonFatal(fallbacks(tsConfig)).leftMap(_.getMessage)
    parsed <- tsConfig.as[A].leftMap(_.show)
```

I like how on each line there is only one thing in scope that has type `Config`.

And as someone reading the code I find it easier to quickly see that each `tsConfig` is used once and then never used again.  As we add more names for things, it takes my brain longer to work out what the code does.